login-frontend/src/App.tsx

```diff
@@ -28,68 +30,136 @@ export function App() {
+            resolve(credentialResponse);
+          });
+          await session.send(credentialRequirements);
+        } catch (e) {
+          reject(e);
+        }
+      });
+
+      let url = '/api/v1/credentials';
+      // get redirect uri query
+      const redirectUri = new URLSearchParams(window.location.search).get('redirect');
+      url = `${url}?redirect=${redirectUri}`;
+
+      const credentialResponseResponse = await fetch(url, {
+        method: 'POST',
+        body: JSON.stringify(credentialResponse),
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        credentials: 'include',
+      });
+
+      if (credentialResponseResponse.status >= 400) {
+        const credentialResponseData = await credentialResponseResponse.text();
+        throw new Error(credentialResponseData);
+      }
+
+      if (credentialResponseResponse.status === 204) {
+        const uri = credentialResponseResponse.headers.get('Location');
+        if (uri !== null) {
+          window.location.href = uri;
+          return;
+        }
+      }
+    },
+    [kilt],
+  );
+
+  const handleSIOPV2Login = useCallback(
+    async (nonce: string, event: FormEvent<HTMLFormElement>) => {
+      connect(process.env.WSS_ENDPOINT ? process.env.WSS_ENDPOINT : 'wss://peregrine.kilt.io');
+
+      const form = event.currentTarget;
+      const extension = new FormData(form).get('extension') as string;
+      const Dids = await kilt[extension].getDidList();
+
+      const did = Dids[0].did;
+
+      const didDocument = await Did.resolve(did);
 
-        if (credentialResponseResponse.status >= 400) {
-          const credentialResponseData = await credentialResponseResponse.text();
-          throw new Error(credentialResponseData);
+      // jwt parts
+      const nbf = Math.floor(Date.now() / 1000);
+      const exp = nbf + 60;
+      const kid = didDocument?.document?.authentication[0].id.replace('#0x', '');
+      const kty = didDocument?.document?.authentication[0].type;
+      // Not completely standard, but SR25519 is not commonly used for JWTs.
+      const header = { alg: 'EdDSA', typ: 'JWT', crv: 'Ed25519', kid, kty };
+      const body = { iss: did, sub: did, nonce, exp, nbf };
+
+      //encoded + signature
+      const headerEncoded = btoa(JSON.stringify(header));
+      const bodyEncoded = btoa(JSON.stringify(body));
+      const dataToSign = headerEncoded + '.' + bodyEncoded;
+
+      const signData = await kilt[extension].signWithDid(sha512.sha512(dataToSign), did);
+
+      // submit token
+      const token = dataToSign + '.' + btoa(signData.signature);
+
+      const url = `/api/v1/did/${token}`;
+
+      const didLoginResponse = await fetch(url, {
+        method: 'POST',
+      });
+
+      if (didLoginResponse.status >= 400) {
+        const credentialResponseData = await didLoginResponse.text();
+        throw new Error(credentialResponseData);
+      }
+
+      if (didLoginResponse.status === 204) {
```

Comment:
all http status codes < 204 are not handled, you could throw an error

Reply:
updated in: [4b81cda](https://github.com/KILTprotocol/opendid/pull/28/commits/4b81cda5a544f6c326a61488174c5ae147bf3fc3)